discovery-provider/src/tasks/entity_manager/social_features.py

```diff
@@ -135,6 +147,60 @@ def delete_social_record(params):
         )
 
 
+def create_social_record_if_access(
+    existing_records: ExistingRecordDict, params: ManageEntityParameters
+):
+    user_id = params.user_id
+    entity_id = params.entity_id
+    entity_type = params.entity_type
+    entity = (
+        existing_records[entity_type][entity_id]
+        if entity_type in existing_records
+        and entity_id in existing_records[entity_type]
+        else None
+    )
+    if entity:
+        premium_content_access = premium_content_access_checker.check_access(
```

Comment:
i think this could be in `validate_social_feature` as well, then create_social_record doesn't need any knowledge of premium access or not